controllers/gatekeeper_controller.go

```diff
@@ -236,14 +237,30 @@ func (r *GatekeeperReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 
 	if err := r.handleCPSController(ctx, gatekeeper); err != nil {
-		if errors.Is(err, errCrdNotReady) {
+		if errors.Is(err, errCpsCrdNotReady) {
 			// ConstraintPodStatus CRD is not ready, wait for the CRD is ready
 			return ctrl.Result{RequeueAfter: time.Second * 3}, nil
 		} else {
 			return ctrl.Result{}, err
 		}
 	}
 
+	if err := r.handleConfigController(ctx); err != nil {
```

Comment:
This is currently blocked if `handleCPSController` errors. Is this intentional?

Reply:
If handleCPSController errors,  restart this reconcile so I thought no issue

Reply:
Right, but there could be a case where `handleCPSController` keeps erroring but it shouldn't block `handleConfigController` in my opinion.

I think the logic should try both but restart the reconcile if either of them error.

Reply:
Should I put in another goroutine?

Reply:
Nah, just call `handleCPSController` and `handleConfigController` and assign the errors to separate variables.

After calling both, then check the errors to determine if you need to requeue.

Reply:
oh I can pass err then at the last I can check err